integration/simulation/simulation.go

```diff
@@ -47,8 +51,22 @@ func NewSimulation(nrNodes int, l1NetworkCfg *L1NetworkCfg, l2NetworkCfg *L2Netw
 		if i == 1 {
 			genesis = true
 		}
+
+		// create an enclave server
+		nodeID := common.BigToAddress(big.NewInt(int64(i)))
+		port := uint64(ENCLAVE_CONN_START_PORT + i)
+		server, err := rpc.StartServer(port, nodeID, stats)
+		if err != nil {
+			panic(fmt.Sprintf("failed to create enclave server: %v", err))
+		}
+		l2NetworkCfg.enclaveServers = append(l2NetworkCfg.enclaveServers, server)
+
 		// create a layer 2 node
-		agg := host.NewAgg(common.BigToAddress(big.NewInt(int64(i))), l2NodeCfg, nil, l2NetworkCfg, stats, genesis)
+		enclaveClient, err := rpc.NewEnclaveClient(port, CLIENT_TIMEOUT_SECS*time.Second)
+		if err != nil {
+			panic(fmt.Sprintf("failed to create enclave client: %v", err))
+		}
+		agg := host.NewAgg(nodeID, l2NodeCfg, nil, l2NetworkCfg, stats, genesis, enclaveClient)
 		l2NetworkCfg.nodes = append(l2NetworkCfg.nodes, &agg)
```

Comment:
Should the `host.NewAgg` and the Enclave setup (client/server)  be tightly coupled ?

Reply:
The server, I don't think so - in the future, the server will be an entirely different process.

Happy to create the client in the constructor.